components/playlist/browser/resources/components/playlistItem.tsx

```diff
@@ -304,7 +307,7 @@ export function PlaylistItem ({
         }
       </ItemInfoContainer>
       <ContextualMenuAnchorButton
-        visible={(hovered || showingMenu) && !isEditing}
+        visible={(hovered || focused || showingMenu) && !isEditing}
```

Comment:
This is 100% optional, but I think you can do most of this with CSS:

```tsx
import PlaylistItemContainer from 'wherever;

const ContextualMenuAnchorButton = styled(Menu)`
    visibility: hidden;

    ${PlaylistItemContainer}:hover & {
        visibility: visible;
    }

    ${PlaylistItemContainer}:focus-within & {
        visibility: visible;
    }

    &[isopen=true] {
        visibility: visible;
    }
```

and you could get rid of the on/off handling here (and completely get rid of `showingMenu` here)

Reply:
That's awesome! Just one thing, contextual menu is used by the header too. But when it's in the header, the button should always be visible. So maybe we should put the CSS in the PlaylistItemContainer side? Also I'm curious how I should handle the `isEditing` case effectively. Do you have any ideas?

Reply:
Whoops completely missed the `isEditing` flag! You could put it in a data attribute on `PlaylistItemContainer`:

```
<PlaylistItemContainer data-editing={isEditing}>
...
</PlaylistItemContainer>
```

and for the CSS something like

```
[data-editing=true] { visibility: hidden !important; }
```

again, completely up to you though - what you have obviously works :smile:

Reply:
Or alternatively you could manually make the selector more specific:

```
[data-editing=true] && { visibility: hidden; }
${PlaylistItemContainer}${PlaylistItemContainer}[data-editing=true] { visibility: hidden; }
```

if you want to avoid a !important

Reply:
Sorry, I tried but CSS version has some minor bugs. I might have done things wrong, but sometimes the button isn't hidden when menu is dismissed. The current version feels more robust to me.